dask/dataframe/shuffle.py

```diff
@@ -454,6 +550,66 @@ def shuffle(
     return df3
 
 
+def _can_use_shuffle_default(
+    shuffle_name: Optional[str], index: Any, df_columns: List[str]
+) -> bool:
+    return True
+
+
+shuffle_registry = ShuffleRegistry()
+
+shuffle_registry.register(
+    priority=0, can_use=_can_use_shuffle_default, method=_shuffle_default
+)
+
+shuffle_registry.register(
+    priority=1,
+    can_use=_can_use_task_shuffle_no_partitions,
+    method=_task_shuffle_no_partitions,
+)
+
+
+def shuffle(
+    df,
+    index,
+    shuffle=None,
+    npartitions=None,
+    max_branch=32,
+    ignore_index=False,
+    compute=None,
+):
+    """Group DataFrame by index
+
+    Hash grouping of elements. After this operation all elements that have
+    the same index will be in the same partition. Note that this requires
+    full dataset read, serialization and shuffle. This is expensive. If
+    possible you should avoid shuffles.
+
+    This does not preserve a meaningful index/partitioning scheme. This is not
+    deterministic if done in parallel.
+
+    See Also
+    --------
+    set_index
+    set_partition
+    shuffle_disk
+    """
+    # Eagerly inspect config for global shuffle selection (if we don't
+    # do this and dask.config.shuffle == "tasks", then we pay the
+    # price of constructing a partitions column even when unnecessary)
+    shuffle = shuffle or config.get("shuffle", None) or "disk"
+    method = shuffle_registry.get(shuffle, index, df.columns)
+    return method(
+        df,
+        index,
+        shuffle=shuffle,
+        npartitions=npartitions,
+        max_branch=max_branch,
```

Comment:
I imagine that there are some key-word arguments that we will want all methods to support, and some that may be method-specific. So, we may want to pass through something like `**kwargs` (and be sure to validate the arguments in the method).